### PR TITLE
Fix undefined reference to some ADNumberTraits static variables.

### LINKED ISSUE #7211: Undefined references in AD classes when compiling with -O0
@jppelteret Would you mind to have a look at http://cdash.kyomu.43-1.org/testDetails.php?test=30048420&build=4774 ? :smiley:

### PR DESCRIPTION
**Pull request by jppelteret**
Fixes #7211

### CODE REVIEW

File: include/deal.II/differentiation/ad/ad_number_traits.h [hunk 1]
```diff
@@ -1009,6 +1009,62 @@ namespace Differentiation
         typename internal::RemoveComplexWrapper<ScalarType>::type,
         ADNumberTypeCode>::n_supported_derivative_levels;
 
+#  else
+
+    template <typename ScalarType, enum NumberTypes ADNumberTypeCode>
+    const bool NumberTraits<
```
Comment (masterleinad) [inline include/deal.II/differentiation/ad/ad_number_traits.h]: This has to be `constexpr` to be consistent with the declaration.

Comment (jppelteret) [reply include/deal.II/differentiation/ad/ad_number_traits.h]: Doh. Although incorrect, this happily compiled locally with GCC 6.3.0 :-/

Comment (jppelteret) [reply include/deal.II/differentiation/ad/ad_number_traits.h]: Thanks for the tip, btw.

**Comment by masterleinad** [comment]
Were you able to reproduce the issue with `gcc-6.3.0`?

**Comment by jppelteret** [comment]
> Were you able to reproduce the issue with gcc-6.3.0?

I must admit that I was in a rush to get this patch out so I didn't check if it failed before I made the changes. Its too late now because my machine is busy again. Perhaps @tamiko would be so kind as to run this on bob to verify that it an effective fix.

**Comment by masterleinad** [comment]
As long as this complies, I am quite positive that it solves the issue indeed.

**Pull request merged**